text/cognito-construct-library/working-backwards-readme.md

```diff
@@ -0,0 +1,300 @@
+user pool ARN or the user pool id. For example:
+
+```ts
+const stack = new Stack(app, 'my-stack');
+
+UserPool.fromUserPoolAttributes(stack, 'my-imported-user-pool', {
+  userPoolId: 'us-east-1_123412341'
+});
+```
+
+> TODO: provider name and provider url return values?
+
+The following sections describe how user pools can be configured. **Unless explicitly stated, all of these
+configurations are available only when creating new user pools.**
+
+### Sign Up
+
+Users need to either signed up by the app's administrators or can sign themselves up. You can read more about both these
+kinds of sign up and how they work
+[here](https://docs.aws.amazon.com/cognito/latest/developerguide/signing-up-users-in-your-app.html).
+
+Further, a welcome email and/or SMS can be configured to be sent automatically once a user has signed up. This welcome
+email and/or SMS will carry the temporary password for the user. The user will use this password to log in and reset
+their password. The temporary password is valid only for a limited number of days.
+
+All of these options can be configured under the `signUp` property. The pool can be configured to let users sign
+themselves up by setting the `selfSignUp` property. A welcome email template can be configured by specifying the
+`welcomeEmail` property and a similar `welcomeSms` property for the welcome SMS. The validity of the temporary password
+can be specified via the `tempPasswordValidity` property.
+
+The user pool can be configured such that either the user's email address, phone number or both should be verifed at the
+time of sign in. Verification is necessary for account recovery, so that there is at least one mode of communication for
+a user to reset their password or MFA token when lost.
+
+When either one or both of these are configured to be verified, a confirmation message and/or email are sent at the
+time of user sign up that they then enter back into the system to verify these attributes and confirm user sign up.
+
+*Note*: If both email and phone number are specified, Cognito will only verify the phone number. To also verify the
+email address, see https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-email-phone-verification.html
+
+*Note*: By default, the email and/or phone number attributes will be marked as required if they are specified as a
+verified contact method. See [attributes](#attributes) section for details about standard attributes.
+
+> TODO: pre-signup lambda trigger
+
+The `verifyContactMethods` attribute allows for this to be configured.
+
+*Defaults*:
+* signUp.selfSignUp: true
+* signUp.tempPasswordValidity: 7 days
+* signUp.welcomeEmail.subject: 'Thanks for signing up'
+* signUp.welcomeEmail.body - 'Hello {username}, Your temporary password is {####}'
+* signUp.welcomeSms.message - 'Your temporary password is {####}'
+* signUp.verifyContactMethods - Email
+
```

Comment:
This is redundant, it is available through the reference documentation and will be easier to keep up to date. Feel free to link to the reference docs.

Reply:
I intend to remove these defaults when this document becomes an actual readme and move them as the `@default` annotation in tsdoc. The `*Defaults*` section is the way to highlight that before the actual code is written.

What's the standard for defaults that we let CloudFormation apply - do we document that as a `@default` in the tsdoc?

Reply:
I've re-worked the defaults in a way that it's not part of the core doc anymore but is still present as a quoted text (`>`). These will go into the tsdoc as `@default` annotations during implementation.